### PR TITLE
Tooltip Upgrades

### LINKED ISSUE #3310: Tooltip
# React
- Fix eslint errors
- Provide a prop to accept a function that runs on toggle.
- Update `triggerIconType` to `triggerIconName`. The icon component will be updated to use name instead of type.

# Angular
- Update `triggerIconType` to `triggerIconName`. The icon component will be updated to use name instead of type.

### PR DESCRIPTION
**Pull request by bobwise**
# React
- Fix eslint errors
- Provide a prop to accept a function that runs on toggle.
- Update `triggerIconType` to `triggerIconName`. The icon component will be updated to use name instead of type.

# Angular
- Update `triggerIconType` to `triggerIconName`. The icon component will be updated to use name instead of type.

# Angular Dictionary
- refactor to not use ngOnInit

### Associated Issue
Fixes #3310 

### Documentation
 - [x] Update Spark Docs

### Code
 - [x] Build Component in Angular
 - [x] Build Component in React
 - [x] Unit Testing in Angular with `npm run test` in `angular/` (100% coverage, 100% passing)
 - [x] Unit Testing in React with `npm run test` in `react/` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)
  
### Accessibility Testing
  - [x] Axe browser extension
  - [x] Jaws Inspect
  - [x] VoiceOver (iOS) or Talkback (Android)

### CODE REVIEW

File: angular/projects/spark-angular/src/lib/components/sprk-dictionary/sprk-dictionary.component.spec.ts [hunk 1]
```diff
@@ -44,71 +74,142 @@ describe('SprkAngularDictionaryComponent', () => {
...
+    wrappedFixture.detectChanges();
+
+    expect(wrappedElement.classList.toString()).toContain(
+      'sprk-c-Dictionary--striped',
+    );
+  });
+
+  it('should prefer keyValuePairs over data', () => {
+    wrappedComponent.keyValuePairs = { key1: 'value1', key2: 'value2' };
+    wrappedComponent.data = { key1: 'value1', key2: 'value2', key3: 'value3' };
```
Comment (RCopeland) [inline angular/projects/spark-angular/src/lib/components/sprk-dictionary/sprk-dictionary.component.spec.ts:200]: is there a warning output if both are supplied? If both are there it means that a component is only partially moved over and [data] is being ignored

File: angular/projects/spark-angular/src/lib/components/sprk-tooltip/sprk-tooltip.component.spec.ts [hunk 2]
```diff
@@ -273,4 +305,56 @@ describe('SprkTooltipComponent', () => {
       tooltipElement.classList.contains('sprk-c-Tooltip--bottom-right'),
```
Comment (RCopeland) [inline angular/projects/spark-angular/src/lib/components/sprk-tooltip/sprk-tooltip.component.spec.ts:305]: missing an expect

File: angular/projects/spark-angular/src/lib/components/sprk-tooltip/sprk-tooltip.component.spec.ts [hunk 3]
```diff
@@ -273,4 +305,56 @@ describe('SprkTooltipComponent', () => {
...
+    wrappedComponent.triggerIconName = 'email';
+    wrappedFixture.detectChanges();
+
+    expect(
+      wrappedTriggerElement.querySelector('use').getAttribute('xlink:href'),
+    ).toEqual('#email');
+  });
+
+  it('should prefer the new property over the deprecated property', () => {
+    wrappedComponent.triggerIconName = 'email';
```
Comment (RCopeland) [inline angular/projects/spark-angular/src/lib/components/sprk-tooltip/sprk-tooltip.component.spec.ts:328]: same here, should output a warning if both are supplied and one is ignored

File: react/src/components/tooltip/SprkTooltip.test.js [hunk 4]
```diff
@@ -205,4 +206,113 @@ describe('SprkTooltip:', () => {
...
+      wrapper.find('button').simulate('click');
+      expect(myMock.mock.calls.length).toBe(0);
+      wrapper.instance().handleWindowClick({});
+      expect(myMock.mock.calls.length).toBe(1);
+    },
+  );
+
+  it('should prefer the new property over the deprecated property', () => {
+    const wrapper = mount(
+      <SprkTooltip
```
Comment (RCopeland) [inline react/src/components/tooltip/SprkTooltip.test.js:289]: should output a warning if both supplied and one is ignored

File: angular/projects/spark-angular/src/lib/components/sprk-tooltip/sprk-tooltip.component.spec.ts (hunk 2 above)
Comment (bobwise) [reply angular/projects/spark-angular/src/lib/components/sprk-tooltip/sprk-tooltip.component.spec.ts:305]: The expect is on the previous line. Prettier reformatted a lot of these lines.

**Pull request merged**